ocaml/tests/test_tar_ext.ml

```diff
@@ -0,0 +1,151 @@
+(*
+ * Copyright (c) Cloud Software Group, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Helpers
+open Tar_ext
+open Bos
+
+let ( // ) = Filename.concat
+
+let gen_test_file_script = "test_data" // "gen_tar_ext_test_file.sh"
+
+let max_size_limit = 2000000L
+
+let create_temp_dir () =
+  let mktemp = Cmd.v "mktemp" in
+  let mktemp' = Cmd.(mktemp % "-d") in
+  let result = OS.Cmd.(run_out mktemp' |> to_string) in
+  match result with
+  | Ok path ->
+      path
+  | Error (`Msg s) ->
+      Alcotest.fail
+        (Printf.sprintf "Test tar_ext creating temp dir failure: %s" s)
```

Comment:
This logic should be put in `gen_tar_ext_test_file.sh` as otherwise it will required every time the `gen_tar_ext_test_file.sh` is called.